dbt/models/iasworld/schema.yml

```diff
@@ -109,21 +134,28 @@ sources:
               name: owndat_unique_by_parid_taxyr
               combination_of_columns:
                 - parid
-                - taxyr      
+                - taxyr
       - name: pardat
         description: '{{ doc("pardat") }}'
         tests:
           - unique_combination_of_columns:
               name: pardat_unique_by_parid_taxyr
               combination_of_columns:
                 - parid
-                - taxyr   
+                - taxyr
       - name: permit
         freshness:
           filter: date_format(date_parse(permdt, '%Y-%m-%d %H:%i:%s.0'), '%Y') >= date_format(current_date - interval '1' year, '%Y')
           warn_after: *48_hours
           error_after: &72_hours {count: 72, period: hour}
       - name: rcoby
       - name: sales
+        tests:
+          - unique_combination_of_columns:
+              name: sales_unique_by_parid_instruno
+              combination_of_columns:
+                - parid
+                - instruno
+              where: substr(saledt, 1, 4) >= '2023'
```

Comment:
**issue (non-blocking):** This conditional hopefully shouldn't be necessary to make the test pass (sales _should_ be unique by document number). If it is, we may need to borrow some of the more complex logic of the sales view, i.e:

```sql
WHERE sales.instruno IS NOT NULL
    AND sales.deactivat IS NULL
    AND tc.township_code IS NOT NULL
```

@wrridgeway Any thoughts here?

Reply:
I don't fully understand the data context here, but I wanted to confirm that the test doesn't pass if you remove the `where` clause:

```
18:07:39  Failure in test sales_unique_by_parid_instruno (models/iasworld/schema.yml)
18:07:39    Got 2133988 results, configured to fail if != 0      
```

Reply:
I'm still curious what @wrridgeway thinks, but I'm going to go ahead and merge in the meantime to unblock test development.

Reply:
for sales, yeah, we can just do unique by doc_no, but we have to include is_multisale = FALSE as well. we should get two rows that fail.

Reply:
@wrridgeway Does that mean we should remove this test and replace it with one like this?

```yaml
- name: sales
  tests:
    - unique_combination_of_columns:
      name: sales_unique_by_doc_no
      combination_of_columns:
        - doc_no
      config:
        where: is_multisale = false
        error_if: ">2
```

Reply:
@jeancochrane That's correct. I'd do:

```yaml
- name: sales
  tests:
    - unique_combination_of_columns:
      name: sales_unique_by_doc_no
      combination_of_columns:
        - doc_no
      config:
        where: not is_multisale
        error_if: ">2
```

Reply:
this is true, but in that case, we do _not_ expect `iasworld.sales` to be unique by `doc_no`. the test i provided jean and which she implemented is the best i can do currently for uniqueness in that table. it _is_ encouraging that for 2023 things are unique by parid and instruno - that's a HUGE improvement. it's also worth noting that i don't think we should _ever_ expect the sales table to be unique only by instruno since multi-pin sales are legitimate.